### PR TITLE
*: implement k8s deployment

### PR DESCRIPTION
**Pull request by you06**
### What problem does this PR solve? 

Implement k8s deployment

### What is changed and how does it work?

```
make tidb
KUBECONFIG=~/.kube/config ./bin/chaos-tidb -namespace test-tidb -image-version v3.0.9
```

### Check List 

Tests 

 - Manual test

Code changes

 - Has Go code change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to update the documentation

### CODE REVIEW

**Comment by you06** [comment]
@mahjonp @mapleFU there are not any comments in the original test-infra, so I do not comment inside it also, PTAL

File: cmd/util/suit.go [hunk 1]
```diff
@@ -53,10 +56,12 @@ func (suit *Suit) Run(ctx context.Context) {
 
 	sctx, cancel := context.WithCancel(ctx)
 
-	err, suit.Config.Nodes = suit.Provisioner.SetUp(sctx, nil)
+	suit.Config.Nodes, err = suit.Provisioner.SetUp(sctx, suit.Cluster)
 	if err != nil {
 		log.Fatalf("deploy a cluster failed, err: %s", err)
 	}
+	log.Println("deploy cluster success")
+	os.Exit(0)
```
Comment (mapleFU) [inline cmd/util/suit.go]: Whats this?

Comment (mahjonp) [inline cmd/util/suit.go]: remove debug stubs?

File: pkg/test-infra/pkg/tidb/recommend.go [hunk 2]
```diff
@@ -58,7 +61,20 @@ func (t *TiDBClusterRecommendation) TiDBReplicas(replicas int32) *TiDBClusterRec
...
+	}
+	b.WriteString(fixture.E2eContext.DockerRepository)
+	b.WriteString("/")
+	b.WriteString(name)
+	b.WriteString(":")
+	b.WriteString(fixture.E2eContext.ImageVersion)
+	return b.String()
+}
+
+// RecommendedTiDBCluster does a recommendation, tidb-operator do not have same defaults yet
```
Comment (mapleFU) [inline pkg/test-infra/pkg/tidb/recommend.go]: Does this publish the svc for tidb cluster?

File: pkg/test-infra/pkg/tidb/recommend.go [hunk 3]
```diff
@@ -58,7 +61,20 @@ func (t *TiDBClusterRecommendation) TiDBReplicas(replicas int32) *TiDBClusterRec
 	return t
 }
 
-// tidb-operator do not have sane defaults yet, do a recommendation
+func buildImage(name string) string {
```
Comment (mapleFU) [inline pkg/test-infra/pkg/tidb/recommend.go]: `buildImage`  is ambiguity...

**Comment by mapleFU** [comment]
Why just have a constexpr set for strings, like tidb/mysql client name

**Comment by you06** [comment]
> Why just have a constexpr set for strings, like tidb/mysql client name

Do you mean this code?

```golang
Ports: []corev1.ServicePort{
	{
		Name: "mysql-client",
		Port: 4000,
	},
	{
		Name: "status",
		Port: 10080,
	},
}
```

File: pkg/test-infra/pkg/tidb/recommend.go (hunk 2 above)
Comment (you06) [reply pkg/test-infra/pkg/tidb/recommend.go]: `*corev1.Service` field in `TiDBClusterRecommendation` will create a nodeport which will public svc for tidb cluster

File: pkg/test-infra/pkg/tidb/recommend.go (hunk 3 above)
Comment (you06) [reply pkg/test-infra/pkg/tidb/recommend.go]: Do you have some advice for the function name?

**Pull request merged**